src/main/java/net/minecraftforge/fmllegacy/packs/PathResourcePack.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.fmllegacy.packs;
+
+import java.nio.file.Path;
+
+public class PathResourcePack extends BasePathResourcePack
```

Comment:
This seems good from my testing. Only comment I'll add here is that this should probably have some documentation that the source passed in references the root folder of the associated nested resource/data pack.

Reply:
Oops yes I forgot to document this class.